sycl/plugins/unified_runtime/pi_unified_runtime.cpp

```diff
@@ -1302,7 +1302,15 @@ __SYCL_EXPORT pi_result piPluginInit(pi_plugin *PluginInit) {
   strncpy(PluginInit->PluginVersion, SupportedVersion, PluginVersionSize);
 
   // Initialize UR and discover adapters
-  HANDLE_ERRORS(urInit(0, nullptr));
+  ur_loader_config_handle_t LoaderConfig{};
+  HANDLE_ERRORS(urLoaderConfigCreate(&LoaderConfig));
+
+  if (PluginInit->SanitizeType == _PI_SANITIZE_TYPE_ADDRESS) {
+    HANDLE_ERRORS(urLoaderConfigEnableLayer(LoaderConfig, "UR_LAYER_ASAN"));
+  }
+
+  HANDLE_ERRORS(urInit(0, LoaderConfig));

```

Comment:
FYI there's an upcoming change where `urInit` is replaced with `urLoaderInit` and `urTearDown` is replaced with `urLoaderTearDown`. I don't think it will be a big change, just want to make you aware that's incoming soon.

Reply:
Thank you for reminding me, @kbenzie